### PR TITLE
fix: colored logs in new logs explorer

### PR DESCRIPTION
**Pull request by vikrantgupta25**
### Summary

This PR takes care of `unescaping` the characters in the logs view where in some receivers ( ex - file log ) the special characters were auto escaped hence no color was being rendered despite the ansi codes being present. 

The same thing has been done for the logs detail view as well where the body can be seen to be rendered with proper colors. 

Also this takes care of removing all the escaped characters while rendering the same in tooltip and body column

#### Related Issues / PR's

contributes to - https://github.com/SigNoz/engineering-pod/issues/1622 

fixes - https://github.com/SigNoz/signoz/issues/4139

#### Screenshots

https://github.com/user-attachments/assets/c31bb203-1df1-406c-9904-d2f9838cedfc

### CODE REVIEW

**Review by nityanandagohain — DISMISSED** [review]

File: frontend/src/components/LogDetail/index.tsx [hunk 1]
```diff
@@ -126,8 +146,8 @@ function LogDetail({
 		>
 			<div className="log-detail-drawer__log">
 				<Divider type="vertical" className={cx('log-type-indicator', logType)} />
-				<Tooltip title={log?.body} placement="left">
-					<Typography.Text className="log-body">{log?.body}</Typography.Text>
+				<Tooltip title={removeEscapeCharacters(log?.body)} placement="left">
+					<div className="log-body" dangerouslySetInnerHTML={htmlBody} />
```
Comment (YounixM) [inline frontend/src/components/LogDetail/index.tsx:150]: @vikrantgupta25 : I remember someone had pointed out the risk of this and we had replied that it must be safe. Just look into that thread to confirm we are not reverting the change.

File: frontend/src/container/LogDetailedView/TableView.tsx [hunk 2]
```diff
@@ -284,19 +290,44 @@ function TableView({
...
+						{record.field === 'body' ? (
+							<CopyClipboardHOC textToCopy={textToCopy}>
+								<span
+									style={{
+										color: Color.BG_SIENNA_400,
+										whiteSpace: 'pre-wrap',
+										tabSize: 4,
+									}}
+									// eslint-disable-next-line react/no-danger
+									dangerouslySetInnerHTML={bodyHtml}
```
Comment (YounixM) [inline frontend/src/container/LogDetailedView/TableView.tsx:315]: Check for any security issue.

File: frontend/src/container/LogDetailedView/utils.tsx [hunk 3]
```diff
@@ -251,18 +251,30 @@ export const getDataTypes = (value: unknown): DataTypes => {
...
-		return escapeMap[char as keyof typeof escapeMap];
-	});
+	str
+		.replace(/\\x1[bB][[0-9;]*m/g, '')
+		.replace(/\\u001[bB][[0-9;]*m/g, '')
+		.replace(/\\x[0-9A-Fa-f]{2}/g, '')
+		.replace(/\\u[0-9A-Fa-f]{4}/g, '')
+		.replace(/\\[btnfrv0'"\\]/g, '');
+
+export const unescapeString = (str: string): string =>
```
Comment (YounixM) [inline frontend/src/container/LogDetailedView/utils.tsx:261]: Add code documentation

File: frontend/src/components/LogDetail/index.tsx (hunk 1 above)
Comment (vikrantgupta25) [reply frontend/src/components/LogDetail/index.tsx:150]: i remember the conversation, we are using dompurify everywhere so no such risk with this. already using this in logs raw and other views

File: frontend/src/container/LogDetailedView/TableView.tsx (hunk 2 above)
Comment (vikrantgupta25) [reply frontend/src/container/LogDetailedView/TableView.tsx:315]: i remember the conversation, we are using dompurify everywhere so no such risk with this. already using this in logs raw and other views

File: frontend/src/container/LogDetailedView/utils.tsx (hunk 3 above)
Comment (vikrantgupta25) [reply frontend/src/container/LogDetailedView/utils.tsx:261]: inline comments added below.. will add more reasoning behind this

Comment (YounixM) [reply frontend/src/container/LogDetailedView/utils.tsx:261]: Yes , let's try adding documentation into the code for better readability.

Comment (vikrantgupta25) [reply frontend/src/container/LogDetailedView/utils.tsx:267]: done ✅

**Review by YounixM — DISMISSED** [review]

**Review by YounixM — APPROVED** [review]

**Review by SagarRajput-7 — APPROVED** [review]

**Pull request merged**